src/pages/TxSender.vue

```diff
@@ -129,7 +134,14 @@ export default defineComponent({
 		formatConfirmedResponse(
 			response: algosdk.modelsv2.PendingTransactionResponse
 		) {
-			this.confirmedResponse = formatJSON(response);
+			// for MyAlgo wallet and Wallet Connect the transaction is not formatted like in Algosigner
+			if (this.walletStore.walletKind === WalletType.ALGOSIGNER) {
+				this.confirmedResponse = formatJSON(response);
+			} else {
+				let formattedJSON: any = response;
+				formattedJSON.txn = prettifyJSON(response.txn);
```

Comment:
I rememer in algorand js sdk we have function (prettyPrint)[https://algorand.github.io/js-algorand-sdk/classes/Transaction.html#prettyPrint].

Can we reuse it?

Reply:
Yeah I checked it, but we are displaying the `EncodedSigned Transaction`, and it returns the `Signed Transaction` and we can't get the encoded transaction from it, also it doesn't encode all keys, only few like `from` and `to` are encoded

Reply:
We can. `SignedTransaction](https://algorand.github.io/js-algorand-sdk/interfaces/SignedTransaction.html) has `txn` field, so you can copy part of this code to return string, and reuse it. I would prefer that.

Reply:
But that txn is `Transaction`, currently we are showing `EncodedTransaction`, both have different keys, so do you want to show `Transaction`..?
Since I checked there code and the encoding function is similar to one I already added

Reply:
Also since we are using the same function to format the confirmed response, that is usually displayed using `EncodedTransaction` instead of `Transaction`

Reply:
I think it would be good to use their code (even copying if needed).
So we are translating from Transaction -> Encoded Transactoin?

Reply:
yes

Reply:
I checked their code and documentation, the encoding is same, it's just that there are specifying the keys and we are rather checking the typeof data to encode the particular key value.
I checked the documentation and looked what all type is possible for different keys of Transaction and I believe that the current code will handle it well.